### ISSUE TITLE
[Bug] catboost integration fails on GPU

### ISSUE THREAD

**Issue by martjushev** [allegroai/clearml#592]
clearml version: 1.2.0
catboost version: 1.0.4
python version: 3.8.10
docker image: nvidia/cuda:11.2.1-cudnn8-runtime-ubuntu20.04

log:
```
Traceback (most recent call last):
  File "catboost_train.py", line 468, in <module>
    main()
  File "catboost_train.py", line 449, in main
    model = train_model(train_params, train_set, test_set)
  File "catboost_train.py", line 265, in train_model
    model.fit(train_set, plot=False, eval_set=test_set, verbose=100)
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/clearml/binding/frameworks/__init__.py", line 36, in _inner_patch
    raise ex
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/clearml/binding/frameworks/__init__.py", line 34, in _inner_patch
    ret = patched_fn(original_fn, *args, **kwargs)
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/clearml/binding/frameworks/catboost_bind.py", line 97, in _fit
    return original_fn(obj, *args, **kwargs)
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/catboost/core.py", line 4768, in fit
    self._fit(X, y, cat_features, text_features, embedding_features, None, sample_weight, None, None, None, None, baseline, use_best_model,
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/catboost/core.py", line 2073, in _fit
    train_params = self._prepare_train_params(
  File "/root/.clearml/venvs-builds/3.8/lib/python3.8/site-packages/catboost/core.py", line 2005, in _prepare_train_params
    _check_train_params(params)
  File "_catboost.pyx", line 5849, in _catboost._check_train_params
  File "_catboost.pyx", line 5867, in _catboost._check_train_params
  File "_catboost.pyx", line 1681, in _catboost._PreprocessParams.__init__
_catboost.CatBoostError: User defined loss functions, metrics and callbacks are not supported for GPU
```

Workaround: 
```
Task.init(
        ...
        auto_connect_frameworks={"catboost": False}  # Doesn't work on a GPU
    )
```

**Comment by eugen-ajechiloae-clearml** [comment]
@martjushev Thanks for reporting this issue! This seems to be an open issue in catboost: https://github.com/catboost/catboost/issues/1792. 
We will patch this such that it doesn't crash on init and look for a way to make metrics reporting work on GPU as well.

**Comment by eugen-ajechiloae-clearml** [comment]
@martjushev We patched this issue in 1.3.0. You should no longer experience a crash, but reporting catboost metrics on GPU will unfortunately not work until user defined callbacks will be supported by catboost.

**Comment by jkhenning** [comment]
Hi @martjushev, closing this. Please re-open if required.
